python/py-virtualenv/Portfile

```diff
@@ -19,24 +19,44 @@ long_description    virtualenv is a tool to create isolated Python \
                     environments.
 
 homepage            https://virtualenv.pypa.io
-master_sites        pypi:v/${python.rootname}
-distname            ${python.rootname}-${version}
 
-checksums           rmd160  1ec5152c8d82350fd24358cfa0d112fdf70e7094 \
-                    sha256  0d62c70883c0342d59c11d0ddac0d954d0431321a41ab20851facf2b222598f3 \
-                    size    5121717
+checksums           rmd160  a3d2e5b06a9bfbc366571961966ccea70a74daca \
+                    sha256  7e4e234e1f27755685dc54063d989756790b0682d60e304db589fa1604938013 \
+                    size    7972183
 
 # keep older Python versions here, do add the EOL version to the list below
 python.versions     27 34 35 36 37 38
 
 if {${name} ne ${subport}} {
     # add EOL warnings
-    if {${python.version} in "34"} {
+    if {${python.version} in "27 34"} {
         PortGroup   deprecated 1.0
+
         deprecated.eol_version  yes
     }
 
-    depends_lib-append  port:py${python.version}-setuptools
+    depends_lib-append  port:py${python.version}-appdirs \
+                        port:py${python.version}-distlib \
+                        port:py${python.version}-filelock \
+                        port:py${python.version}-setuptools \
```

Comment:
doesn't depend on setuptools 👍

Reply:
Shouldn't  all the python ports depend on setuptools?

Reply:
nope 👍 only old-style setup.py builds (which are not supported anymore) so hope we don't do that... if you build via pip or pep-517 this should not be needed

Reply:
Mh, could you please link me an example of a portfile using `pep-517` or `pip` to install? I tried to grep around but I didn't find anything useful... Or it's automatic with the PG?

Reply:
we do build with ```setup.py``` so the build- and runtime dependency on ```setuptools``` is needed here

Reply:
setuptools does not support this behaviour anymore, other than backwards compatible way, so you might probably want to change to pip and wheels 🤷‍♂️

Reply:
virtualenv itself uses pep 517 and 518 standards to build itself, so I don't think you'll be able to build without provisioning first the build system per pep-518